Library/Formula/gobject-introspection.rb

```diff
@@ -17,6 +29,7 @@ class GobjectIntrospection < Formula
   depends_on "cairo"
   depends_on "libffi"
   depends_on "python" if MacOS.version <= :mavericks
+  depends_on "python3" => :optional
```

Comment:
Just put this in the `head` block, then you can bin the `if build.with?("python3") && !build.head?` logic and this line. Also `:python3`, not `"python3"`.